### DISCUSSION TITLE
How to display docstring output for function with multiple methods?

### DISCUSSION THREAD

**Discussion by nathanrboyer** [fonsp/Pluto.jl#3103] [Questions about using Pluto] [answered]
I like the feature where a function defined in a cell shows its docstring as the output. However, I am having trouble getting that to work when multiple methods are required.

The [Pluto FAQ](https://github.com/fonsp/Pluto.jl/wiki/%E2%9A%A1-Writing-and-running-code) says to write multiple methods in a begin-end block, but then I get no output.

![image](https://github.com/user-attachments/assets/a8d4ec4b-384c-4e4d-8afc-eefe779629e0)
![image](https://github.com/user-attachments/assets/3ed5fa05-37cc-42b3-9ba2-0166ca22a6da)

Contrary to the FAQ information, the method definitions seem to work okay in separate cells, but then I get a second output. I can suppress the second output with `;`, but this puts a weird gap in my sequence of docstrings. Does the FAQ information need to be updated?

![image](https://github.com/user-attachments/assets/1bfc20e1-2d25-4f53-88f5-a3454f12f98c)

**Comment by fonsp** [comment, marked as answer]
Hey! There is no obvious way to do it, but this what I do:

<img width="549" alt="image" src="https://github.com/user-attachments/assets/075b5c11-f3a7-49cc-925e-330efb9c4f69">


```julia
begin
	f(x::String) = x
	f(x::Int) = x + 1

	@doc """
	```julia
	f(x::Union{String,Int})
	```

	Return something like `x`.
	
	""" f

end
```

---

Maybe you could also write your own macro? But docstrings are a bit tricky to work with in macros.

**Comment by nathanrboyer** [comment]
It still works without the explicit `@doc` macro. You just have to have the documented function be the last item in the `begin` block.
![image](https://github.com/user-attachments/assets/471e46a2-4b7e-4c95-938f-e261d37e7c83)
